addons/portal_rating/models/portal_rating.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class PortalRatingMixin(models.AbstractModel):
+    """Technical mixin that unite portal and rating mixin to solve inheritance problem."""
```

Comment:
Good idea, by doing it, I now better understand the problem. I let you read the explanation and if I have well understood, I think it could potentially cause more problem in the future because if mail_thread got sorted first in the inheritance chain, all override of mail_thread method in parent module could be ignored. Is that right ?